controller/users.go

```diff
@@ -49,6 +50,119 @@ func NewUsersController(service *goa.Service, db application.DB, config UsersCon
 		config:     config,
 	}
 }
+func randString(length int) string {
+	const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	randomStr := make([]byte, length)
+	for i := range randomStr {
+		randomStr[i] = letterBytes[rand.Int63()%int64(len(letterBytes))]
+	}
+	return string(randomStr)
+}
+
+// Obfuscate runs the obfuscate action to invalidate the sensitive data associated
+// with an user and her associated identity.
+func (c *UsersController) Obfuscate(ctx *app.ObfuscateUsersContext) error {
+	isSvcAccount, err := isServiceAccount(ctx, serviceNameAuth)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to determine if account is a service account")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err))
+
+	}
+	if !isSvcAccount {
+		log.Error(ctx, map[string]interface{}{
+			"identity_id": ctx.ID,
+		}, "account used to call obfuscate API is not a service account")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(errs.New("account used to call obfuscate API is not a service account")))
+	}
+	u, err := uuid.FromString(ctx.ID)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to convert user id in valid uuid")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrBadRequest(errs.New("invalid user id")))
+
+	}
+	err = application.Transactional(c.db, func(appl application.Application) error {
+		obfStr := randString(12)
+
+		// Obfuscate User
+		users, err := appl.Users().Query(account.UserFilterByID(u))
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"user_id": u,
+				"err":     err,
+			}, "unable to load the user")
+			return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err))
```

Comment:
1. Please use `errors.NewNotFoundError` and not `goa.Err...` ever.
2. You must not use `jsonapi.JSONErrorResponse` within a `application.Transactional` because it will return `ctx.NotFound(jsonErr)` which itself returns `nil` if I'm not mistaken. @xcoulon had raised this PR #1955 explicitly to address this.